### PR TITLE
Add nil checks to `_inherent()`

### PR DESCRIPTION
**Pull request by mamekoro**
The LSP specification states that `CompletionItem.labelDetails` can be nil, so it is better to check that `labelDetails` is not nil before accessing `labelDetails.detail` in `_inherent()`.

The use of `label` was eliminated in the [previous PR](https://github.com/zjp-CN/nvim-cmp-lsp-rs/pull/10), but has been restored in this PR.

This is because I found that rust-analyzer assigns `item.label_detail`, which may contain a string such as `"(as Trait)"` or `"(use Import)"`, to `lsp_item.label` instead of `lsp_item.label_details.detail` if the LSP client does not support the labelDetails capability. ([source code](https://github.com/rust-lang/rust-analyzer/blob/fc98e0657abf3ce07eed513e38274c89bbb2f8ad/crates/rust-analyzer/src/lsp/to_proto.rs#L368-L379))

In other words, we need to handle `label` to support LSP clients that lack support for the labelDetails capability.

### CODE REVIEW

**Comment by zjp-CN** [comment]
:heart: Really appreciate your PR and the comments!
